Simbody/src/CableSpan.cpp

```diff
@@ -0,0 +1,2354 @@
+
+const CurveSegment* CableSpan::Impl::findPrevActiveCurveSegment(
+    const State& s,
+    ObstacleIndex ix) const
+{
+    for (int i = ix - 1; i >= 0; --i) {
+        // Find the active segment before the current.
+        if (m_CurveSegments.at(ObstacleIndex(i))
+                .getInstanceEntry(s)
+                .isInContactWithSurface()) {
+            return &m_CurveSegments.at(ObstacleIndex(i));
+        }
+    }
+    return nullptr;
+}
+
+const CurveSegment* CableSpan::Impl::findNextActiveCurveSegment(
+    const State& s,
+    ObstacleIndex ix) const
+{
+    // Find the active segment after the current.
+    for (int i = ix + 1; i < m_CurveSegments.size(); ++i) {
+        if (m_CurveSegments.at(ObstacleIndex(i))
+                .getInstanceEntry(s)
+                .isInContactWithSurface()) {
+            return &m_CurveSegments.at(ObstacleIndex(i));
+        }
+    }
+    return nullptr;
+}
+
+Vec3 CableSpan::Impl::findPrevPoint(const State& s, ObstacleIndex ix) const
+{
+    auto CalcCurveFinalContactPoint = [&](const CurveSegment& curve) -> Vec3 {
+        const Transform& X_GS           = curve.calcSurfaceFrameInGround(s);
+        const Vec3& finalContactPoint_S = curve.getInstanceEntry(s).X_SQ.p();
+        return X_GS.shiftFrameStationToBase(finalContactPoint_S);
+    };
+
+    const CurveSegment* prevCurve = findPrevActiveCurveSegment(s, ix);
+    return prevCurve
+               ? CalcCurveFinalContactPoint(*prevCurve)
+               : getOriginBody().getBodyTransform(s).shiftFrameStationToBase(
+                     m_OriginPoint);
+}
+
+Vec3 CableSpan::Impl::findNextPoint(const State& s, ObstacleIndex ix) const
+{
+    auto CalcCurveInitialContactPoint = [&](const CurveSegment& curve) -> Vec3 {
+        const Transform& X_GS             = curve.calcSurfaceFrameInGround(s);
+        const Vec3& initialContactPoint_S = curve.getInstanceEntry(s).X_SP.p();
+        return X_GS.shiftFrameStationToBase(initialContactPoint_S);
+    };
+
+    const CurveSegment* nextCurve = findNextActiveCurveSegment(s, ix);
+    return nextCurve ? CalcCurveInitialContactPoint(*nextCurve)
+                     : getTerminationBody()
+                           .getBodyTransform(s)
+                           .shiftFrameStationToBase(m_TerminationPoint);
+}
+
+void CableSpan::Impl::callForEachActiveCurveSegment(
+    const State& s,
+    std::function<void(const CurveSegment&)> f) const
+{
+    for (const CurveSegment& curve : m_CurveSegments) {
+        // Skip non-active segments.
+        if (!curve.getInstanceEntry(s).isInContactWithSurface()) {
+            continue;
+        }
+        // Call provided function for the active segments.
+        f(curve);
+    }
+}
+
+//==============================================================================
+//                      CABLE FORCE COMPUTATIONS
+//==============================================================================
+
+namespace
```

Comment:
Stopped here (for now)

Reply:
@adamkewley thank you for all the comments!
I think I resolved them, but let me know what you think!

A note on the formatting, I looked around at other files and saw different styles regarding banners and naming. (I liked the more recent `ExponentialSpringForce.cpp/h`) I tried to clean it up, and use the following rules:
- No banners in header `CableSpan.h`
- A loose approach with Banners & namespaces in `CableSpan.cpp` to help reading it
- `m_xxx` naming for fields of classes, but not for fields of structs

(I found examples and counter examples for the above, so I tried to get to something that helps understand the code)

Would be great if you wanted to have another go :)